mne/utils.py

```diff
@@ -3172,3 +3172,354 @@ def _check_if_nan(data, msg=" to be plotted"):
+
+            2. ``epochs['name']``: Return ``Epochs`` object with a copy of the
+               subset of epochs corresponding to an experimental condition as
+               specified by 'name'.
+
+               If conditions are tagged by names separated by '/' (e.g.
+               'audio/left', 'audio/right'), and 'name' is not in itself an
+               event key, this selects every event whose condition contains
+               the 'name' tag (e.g., 'left' matches 'audio/left' and
+               'visual/left'; but not 'audio_left'). Note that tags selection
+               is insensitive to order: tags like 'auditory/left' and
+               'left/auditory' will be treated the same way when accessed.
+
+            3. ``epochs[['name_1', 'name_2', ... ]]``: Return ``Epochs`` object
+               with a copy of the subset of epochs corresponding to multiple
+               experimental conditions as specified by
+               ``'name_1', 'name_2', ...`` .
+
+               If conditions are separated by '/', selects every item
+               containing every list tag (e.g. ['audio', 'left'] selects
+               'audio/left' and 'audio/center/left', but not 'audio/right').
+
+            4. ``epochs['pandas query']``: Return ``Epochs`` object with a
+               copy of the subset of epochs (and matching metadata) that match
+               ``pandas query`` called with ``self.metadata.eval``, e.g.::
+
+                   epochs["col_a > 2 and col_b == 'foo'"]
+
+               This is only called if Pandas is installed and ``self.metadata``
+               is a :class:`pandas.DataFrame`.
+
+               .. versionadded:: 0.16
+        """
+        return self._getitem(item)
+
+    def _getitem(self, item, reason='IGNORED', copy=True, drop_event_id=True,
+                 select_data=True, return_indices=False):
+        """
+        Select epochs from current object.
+
+        Parameters
+        ----------
+        item: slice, array-like, str, or list
+            see `__getitem__` for details.
+        reason: str
+            entry in `drop_log` for unselected epochs
+        copy: bool
+            return a copy of the current object
+        drop_event_id: bool
+            remove non-existing event-ids after selection
+        select_data: bool
+            apply selection to data
+            (use `select_data=False` if subclasses do not have a
+             valid `_data` field)
+        return_indices: bool
+            return the indices of selected epochs from the original object)
+            in addition to the new `Epochs` objects
+        Returns
+        -------
+        `Epochs` or tuple(Epochs, np.ndarray) if `return_indices` is True
+
+        object with subset of epochs (and optionally array with kept
+        epoch indices)
+        """
+        data = self._data
+        del self._data
+        inst = self.copy() if copy else self
+        self._data, inst._data = data, data
```

Comment:
`self._data = inst._data = data`?

Reply:
This and a few of the other things you noted are older code that I moved around so that it could be used by both Epochs and EpochsTFR. I don't like to change things too much, in case I am missing why they needed to be this way. But this is simple enough, thanks for pointing it out.

Reply:
Yeah I saw eventually that code was just moved around ... but still, if it's cleaner afterwards, even better!